tests/testthat/test-utils.R

```diff
@@ -122,9 +121,51 @@ test_that("strict_mode_error() works correctly [BBR-UTL-010]", {
   })
 })
 
-
 test_that("suppressSpecificWarning() works [BBR-UTL-011]", {
   # log() of a negative number raises a warning
   x <- suppressSpecificWarning(log(-1), "NaNs produced")
   expect_true(is.nan(x))
 })
+
+test_that("warning raised when threads > 1 and parallel is FALSE [BBR-UTL-014]", {
+  withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()),{
+    withr::with_tempdir({
+      fs::dir_create(file.path(tempdir(), "test_path"))
+
+      on.exit(if(fs::dir_exists(file.path(tempdir(),"test_path")))
+      {
+        fs::dir_delete(file.path(tempdir(),"test_path"))
+      })
+
+      files_to_copy <- file.path(ABS_MODEL_DIR, c("1.ctl"))
+
+      fs::file_copy(system.file("extdata", "acop.csv", package = "bbr"), file.path(tempdir(), "test_path"))
+      purrr::walk(files_to_copy, fs::file_copy, file.path(tempdir(), "test_path"))
+      fs::dir_copy(file.path(ABS_MODEL_DIR, "1"), file.path(tempdir(), "test_path"))
+      ctl <- read_lines(file.path(tempdir(),"test_path", "1.ctl")) %>%  stringr::str_remove("../../../../extdata/")
+      write_lines(ctl, file.path(tempdir(),"test_path", "1.ctl"))
+
+      mod1 <- new_model(file.path(tempdir(), "test_path", "1"), .description = "original test-workflow-bbi model",
+                        .tags = ORIG_TAGS,.bbi_args = list(overwrite = TRUE, threads = 2))
+
+      readr::write_file("created_by: test-utils", file.path(tempdir(), "test_path", "bbi.yaml"))
+
+
+      res <- capture.output(submit_model(mod1, .dry_run = TRUE))
+
+      #Testing that check_bbi_args is appending --parallel when not passed
+      expect_identical("--parallel", str_subset(res,"--parallel") %>% str_extract("--parallel"))
+      fs::file_delete(file.path(tempdir(), "test_path", "1.yaml"))
```

Comment:
I think this test right here addressed one of my previous concerns (if threads > 1 and parallel not set: parallel is set to TRUE). 

I would change the test description, as well as the description in the requirements yaml to be more broad, since we're technically testing for more than just the warning here. I think its fine to add all new tests (warning, setting parallel to TRUE, and leaving parallel alone) all to the same requirement you made, as long as the description encompasses those things. 

Could be something like:

```yaml
UTL-R014:
  description: when `threads > 1`, check_bbi_args() warns if parallel is set to `FALSE`, and automatically sets parallel to `TRUE` if not specified.
  tests:
  - BBR-UTL-014 (includes two tests)
  - BBR-UTL-015 (new regression test - see above comment)
```

Reply:
I was trying to keep this within one test because of all the set-up required. Is it alright to duplicate the code?

Reply:
Yeah I take your point. One option is to copy all those files over as part of the setup (see the first 34 lines of `tests/testthat/test-test-threads.R`), but you could include the additional test as part of that (up to you). 

Either way I would update the **requirement** **description** to be more general (since its not just about the warning). If you do choose to include all 3 tests under the same **test** **description**, I would make that one more general too